### ISSUE TITLE
Gyro spontaneous output, messes up aiming

### ISSUE THREAD

**Issue by VoxelTek** [XITRIX/Moonlight-Switch#169]
Hi there,

When using gyro, I get spontaneous and unprompted input. I feel the best way to show this is with a demonstration.

https://github.com/XITRIX/Moonlight-Switch/assets/53562267/16837586-5fb3-4653-a24f-12d3791d8bdc

[hl2.webm](https://github.com/XITRIX/Moonlight-Switch/assets/53562267/277b798f-8644-4ef6-9ba8-08568318b5e3)

As you can see, the gyro input is not stable. The jittering is very, very annoying, and makes pretty much every game impossible to play using it.

I am sure that this isn't a problem with my Switch itself or my PC, as using another PC as a client and running gyro input on that works fine, and gyro aiming functions fine in Switch games.

If I am doing something wrong, please let me know, and I'll attempt to fix the issue.

Kind regards,
@voxeltek

**Comment by XITRIX** [comment]
Yes, that's looks really bad and definitely should not behave like that, I'll try to check if I can reproduce that.

Just in case, which version of you have installed?

**Comment by VoxelTek** [comment]
Hey! Thanks for investigating, I'm currently running Moonlight-Swirch V1.1.0.

**Comment by XITRIX** [comment]
I have not enough time to work on it right now, but I'll pin it so when I'll have some time I will not forget about it

**Comment by markandgo** [comment]
Not sure if this is related but I am using Sunshine v0.23.1 with Moonlight v1.2.1, and I notice that the motion control control drifts over use/time. Turning left and right would eventually reverse, e.g. left would turn right. Turning up and down would become diagonally up or down.

**Comment by vrisalab** [comment]
I can confirm this issue also happens to me to a lesser extent, with snapping ocurring about once or twice per minute (Enough to be very annoying).
It happens to me with moonlight linux with a ds4 controller too, with the same sunshine  v2025.310.123250 instance, and using Steam Input.

**Comment by VoxelTek** [comment]
> It happens to me with moonlight linux with a ds4 controller too, with the same sunshine v2025.310.123250 instance, and using Steam Input.
@vrisalab

Sounds like this might be a different problem, then. My issue was exclusively when using the Switch, and didn't occur when using Sunshine Linux and Moonlight Linux, with a regular Switch Pro controller. I haven't tested for a while, though, so it's possible something's broken since then.

**Comment by vrisalab** [comment]
I've gone back and done some testing, and now the exact same thing it's happening to me, with the same frequency. I'm not sure what has changed.
It's likely then that my moonlight linux issue is completely separate, judging from what you say.

**Comment by XITRIX** [comment]
I tested gyro for a bit, I cannot reproduce this jumps on my Windows Sunshine host.
I will not say that I’m 100% sure it’s not a Client issue, but could you try to doublecheck, that it’s not happening with any other Moonlight client (Android i.e.)

**Comment by VoxelTek** [comment]
> I tested gyro for a bit, I cannot reproduce this jumps on my Windows Sunshine host. I will not say that I’m 100% sure it’s not a Client issue, but could you try to doublecheck, that it’s not happening with any other Moonlight client (Android i.e.)

I currently am unable to test, but while I was, I tried using Moonlight on another machine, and using a gyro controller there worked fine. There's also a significant number of people using gyro on Sunshine Linux, and they do not experience this issue.
